test/config-luatest/config_info_hierarchy_test.lua

```diff
@@ -0,0 +1,153 @@
+local fio = require('fio')
+local t = require('luatest')
+local cbuilder = require('luatest.cbuilder')
+local cluster = require('test.config-luatest.cluster')
+local server = require('luatest.server')
+local it = require('test.interactive_tarantool')
+
+local g = t.group()
+
+g.before_all(cluster.init)
+g.after_each(cluster.drop)
+g.after_all(cluster.clean)
+
+g.before_test('test_script', function(g)
+    g.server = server:new({alias = 'classic_server'})
+    g.server:start()
+end)
+
+g.after_test('test_script', function(g)
+    g.server:stop()
+    g.server = nil
+end)
+
+g.before_test('test_interactive', function(g)
+    g.it = it.new()
+end)
+
+g.after_test('test_interactive', function(g)
+    g.it:close()
+    g.it = nil
+end)
+
+local function assert_configured_names(server)
+    server:exec(function()
+        local config = require('config')
+
+        local exp = {
+            group = 'g-001',
+            replicaset = 'r-001',
+            instance = 'i-001',
+        }
+
+        t.assert_equals(config:info().hierarchy, exp)
+        t.assert_equals(config:info('v1').hierarchy, exp)
+        t.assert_equals(config:info('v2').hierarchy, exp)
+        t.assert_equals(box.info.config.hierarchy, exp)
+    end)
+end
```

Comment:
Nit: probably, it's worth use something like `assert_config_info_hierarchy` accepting the server and the expected value. This function seems more versatile.

The expected value could be optional with the default value equal to existing local `exp` in `assert_configured_names`

Reply:
Implemented.